pydantic/fields.py

```diff
@@ -401,7 +401,7 @@ def _get_field_info(
             field_info = next(iter(field_infos), None)
             if field_info is not None:
                 field_info.update_from_config(field_info_from_config)
-                if field_info.default is not Undefined:
+                if field_info.default not in (Undefined, Required):
```

Comment:
```suggestion
                if field_info.default not in {Undefined, Required}:
```
Weirdly I've seen in the past that checking containment in sets can be faster even when they're tiny.

Reply:
But if you're not sure, might be worth checking.

Reply:
Some napkin testing on an i7 2020 macbook, homebrew 3.9 python:
```
from timeit import timeit
from enum import Enum
class TestEnum(Enum):
    Undefined = 1
    Required = 2
    Whatever = 3
def test():
    [a not in (TestEnum.Undefined, TestEnum.Required) for a in TestEnum]
def test_set():
    [a not in {TestEnum.Undefined, TestEnum.Required} for a in TestEnum]
timeit(test)
2.068943857000022
timeit(test_set)
3.550850099999934
```

Now granted this is not super accurate (not the same enum size, testing all options in order instead of sorting the most common tests, etc) but for the purposes of what we're checking, testing over what's effectively an array of size 2 is quite faster, vs finding hashes of all members, storing them, etc which is what you'd expect to happen with the set option.